File: compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/Hash.hs
-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
-- SPDX-License-Identifier: Apache-2.0

module DA.Daml.LF.Proto3.Hash where

import           DA.Prelude
import qualified Data.ByteString          as BS
import qualified Data.Text                as T
import qualified Numeric

-- | Encode the hash bytes of the payload in the canonical
-- lower-case ascii7 hex presentation.
encodeHash :: BS.ByteString -> T.Text
encodeHash = T.pack . reverse . foldl' toHex [] . BS.unpack
  where
    toHex xs c =
      case Numeric.showHex c "" of
        [n1, n2] -> n2 : n1 : xs
        [n2]     -> n2 : '0' : xs
        _        -> error "impossible: showHex returned [] on Word8"

File: sdk/canton/community/app-base/src/main/scala/com/digitalasset/canton/admin/api/client/data/crypto/Hash.scala
// Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.digitalasset.canton.admin.api.client.data.crypto

import com.digitalasset.canton.logging.pretty.{Pretty, PrettyPrinting}

sealed abstract class HashAlgorithm(val name: String) extends PrettyPrinting {
  override val pretty: Pretty[this.type] = prettyOfString(_.name)
}

object HashAlgorithm {
  case object Sha256 extends HashAlgorithm("SHA-256")
}
